### PR TITLE
Fix some argument validation issues in fetch()

### PR DESCRIPTION
**Pull request by Jarred-Sumner**
### What does this PR do?

Fix some argument validation issues in fetch()

Fixes https://github.com/oven-sh/bun/issues/12764
Fixes https://github.com/oven-sh/bun/issues/12483

### How did you verify your code works?

There are tests

### CODE REVIEW

**Comment by robobun** [comment]
@Jarred-Sumner, your commit 2518c6a0c3e29aa50e775f2d7587391865805fdd is building: [`#1946`](https://buildkite.com/bun/bun/builds/1946)

File: src/bun.js/api/server.zig [hunk 1]
```diff
@@ -5498,6 +5498,10 @@ pub fn NewServer(comptime NamespaceType: type, comptime ssl_enabled_: bool, comp
                         data_value = headers_value;
                     }
 
+                    if (globalThis.hasException()) {
```
Comment (cirospaciari) [inline src/bun.js/api/server.zig:5501]: in the future should methods that can throw return a `error.JSException` so we are forced to check this.

File: src/bun.js/bindings/bindings.cpp [hunk 2]
```diff
@@ -4254,7 +4285,7 @@ static void populateStackFramePosition(const JSC::StackFrame* stackFrame, BunStr
         // It is key to not clone this data because source code strings are large.
         // Usage of toStringView (non-owning) is safe as we ref the provider.
         provider->ref();
-        ASSERT(*referenced_source_provider == nullptr);
+        // ASSERT(*referenced_source_provider == nullptr);
```
Comment (cirospaciari) [inline src/bun.js/bindings/bindings.cpp]: is this intencional?

File: src/bun.js/webcore/request.zig [hunk 3]
```diff
@@ -777,22 +777,39 @@ pub const Request = struct {
...
+            this._headers = FetchHeaders.createFromUWS(req);
         } else {
             // we don't have a request context, so we need to create an empty headers object
             this._headers = FetchHeaders.createEmpty();
 
             if (this.body.value == .Blob) {
                 const content_type = this.body.value.Blob.content_type;
                 if (content_type.len > 0) {
-                    this._headers.?.put("content-type", content_type, globalThis);
+                    this._headers.?.put(.ContentType, content_type, globalThis);
```
Comment (cirospaciari) [inline src/bun.js/webcore/request.zig:796]: nice

File: src/bun.js/webcore/response.zig [hunk 4]
```diff
@@ -2030,572 +2045,662 @@ pub const Fetch = struct {
...
-                if (hostname) |host| {
-                    allocator.free(host);
-                    hostname = null;
-                }
+            break :extract_url bun.String.empty;
+        };
+        defer url_str.deref();
+
+        if (globalThis.hasException()) {
+            is_error = true;
```
Comment (cirospaciari) [inline src/bun.js/webcore/response.zig:2136]: I feel like this is_error is in a lot f places may we should refactor in the future to return zig errors and errdefer and handle the JS exception in a different way

**Review by cirospaciari — APPROVED** [review]
LGTM

File: src/bun.js/webcore/response.zig (hunk 4 above)
Comment (Jarred-Sumner) [reply src/bun.js/webcore/response.zig:2136]: yeah

**Pull request merged**